src/PuliExtension.php

```diff
@@ -99,7 +100,7 @@ public function getTokenParsers()
     public function getFunctions()
     {
         if (!$this->urlGenerator) {
-            return array();
+            throw new RuntimeException('The resource_url() Twig function requires a UrlGenerator to be set in PuliExtension.');
```

Comment:
I see :) The proper solution seems to be to simply remove the if statement. Please add a comment here that the exception is thrown in TemplatePathResolver so that it is clear that we don't check for the presence of `urlGenerator` here by intention.